spec/std/io/memory_spec.cr

```diff
@@ -226,6 +226,44 @@ describe IO::Memory do
     io.gets(1).should eq("d")
   end
 
+  it "can truncate to size" do
+    io = IO::Memory.new
+    io << "hello world"
+    io.truncate(5)
+    io.to_s.should eq("hello")
+  end
+
+  it "raises when truncate size is too large" do
+    io = IO::Memory.new
+    io << "hello world"
+
+    io.truncate(12)
+    io.truncate(13)
+
+    io.to_s.should eq("hello world\u0000\u0000")
```

Comment:
Should truncating with `size > bytesize` really fill up with zero bytes?
This is not truncating but expanding and the semantic difference is only governed by the current bytesize.
I'd rather have it fault tolerant and don't grow bytesize on truncate.

Reply:
`bytesize` growing on truncate matches the behavior of `File`.